File: frontend/src/api/sheets.ts
import apiClient from './client';
import type { GoalSheet } from '../types';

export const getMySheet = async () => {
  const response = await apiClient.get<GoalSheet>('/sheets/me');
  return response.data;
};

export const createSheet = async (cycleId: string) => {
  const response = await apiClient.post<GoalSheet>('/sheets', { cycle_id: cycleId });
  return response.data;
};

export const submitSheet = async (sheetId: string) => {
  const response = await apiClient.post(`/sheets/${sheetId}/submit`);
  return response.data;
};

export const getTeamSheets = async () => {
  const response = await apiClient.get<GoalSheet[]>('/sheets/team');
  return response.data;
};

export const approveSheet = async (sheetId: string) => {
  const response = await apiClient.patch(`/sheets/${sheetId}/approve`);
  return response.data;
};

export const returnSheet = async (sheetId: string, comment: string) => {
  const response = await apiClient.patch(`/sheets/${sheetId}/return`, { comment });
  return response.data;
};

File: backend/routers/sheets.py
from fastapi import APIRouter, Depends, Body
from typing import List
from core.auth import get_current_user, check_role
from services.sheet_service import sheet_service
from models.sheet import GoalSheetCreate

router = APIRouter(prefix="/sheets", tags=["Sheets"])

@router.post("")
async def create_sheet(data: GoalSheetCreate, user: dict = Depends(get_current_user)):
    return sheet_service.create_sheet(user["id"], data.cycle_id)

@router.get("/me")
async def get_my_sheet(user: dict = Depends(get_current_user)):
    return sheet_service.get_my_sheet(user["id"])

@router.post("/{sheet_id}/submit")
async def submit_sheet(sheet_id: str, user: dict = Depends(get_current_user)):
    return sheet_service.submit_sheet(sheet_id, user["id"])

@router.get("/team")
async def get_team_sheets(user: dict = Depends(check_role(["manager", "admin"]))):
    return sheet_service.get_team_sheets(user["id"])

@router.patch("/{sheet_id}/approve")
async def approve_sheet(sheet_id: str, user: dict = Depends(check_role(["manager", "admin"]))):
    return sheet_service.approve_sheet(sheet_id, user["id"])

@router.patch("/{sheet_id}/return")
async def return_sheet(sheet_id: str, comment: str = Body(..., embed=True), user: dict = Depends(check_role(["manager", "admin"]))):
    return sheet_service.return_sheet(sheet_id, user["id"], comment)
